File: April/Abiji-2020/Golang /Sum of Left Leaves/Sum of Left Leaves.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumOfLeftLeaves(root *TreeNode) int {

    if root == nil{
        return 0
    }
    sum:= 0
    if (root.Left!=nil && root.Left.Left==nil && root.Left.Right == nil){
        sum+=root.Left.Val
    }
    sum+=sumOfLeftLeaves(root.Left)+sumOfLeftLeaves(root.Right)
    return sum
    
}

File: April/Abiji-2020/C++/Sum of Left Leaves/Sum of Left leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {

public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(root == nullptr) return 0;
        int sum =0;
        if(root->left!=nullptr && root->left->left==nullptr && root->left->right==nullptr) sum+=root->left->val;
        sum+= sumOfLeftLeaves(root->left)+sumOfLeftLeaves(root->right);
        return sum;
    }
};
